### ISSUE TITLE
Formatting time_point<utc/tai_clock, seconds>::min/max() yields unexpected string

### ISSUE THREAD

**Issue by marshray** [HowardHinnant/date#303]
With the following code:
```
#include <iostream>
#include <chrono>

#include "date/date.h"
#include "date/tz.h"

using namespace std;
using namespace std::chrono;

using utc_point = time_point<date::utc_clock, std::chrono::seconds>;
using tai_point = time_point<date::tai_clock, std::chrono::seconds>;

int main(int, char *[])
{
    cout
    << "utc_point min, max\n"
    << utc_point::min() << "\n"
    << utc_point::max() << "\n\n"
    << "tai_point min, max\n"
    << tai_point::min() << "\n"
    << tai_point::max() << "\n";
    return 0;
}
```
I would expect that the resulting formatted values would:
A. represent the 'hour' field as two digits between '00' to '23' and
B. the min() value would refer to a date earlier than that of the max() value.

Instead I obtain the following results:
```
$ clang++ -DUSE_OS_TZDB=1 -DHAS_REMOTE_API=0 -std=c++14 -I../../include -I../../include/date -Wall ../../src/tz.cpp main.cpp && ./a.out && echo ok
utc_point min, max
-10484-10-16 -2562033891409911:30:08
14423-03-17 2562033891409935:29:40

tai_point min, max
14411-03-17 2562033891409935:30:08
14411-03-17 2562033891409935:30:07
ok
```

A. Both utc and tai_points print an out-of-range value for the 'hour' field.
B. The date portion of the UTC values appear plausible, but with TAI both the min() and max() values refer to the same date. Furthermore, the time portion of min() appears later than that of max().

G++ and clang++ produce identical output.

Thanks,

- Marsh

```
$ g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.2.0-8ubuntu3.2' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.2.0 (Ubuntu 7.2.0-8ubuntu3.2)
```

```
$ clang++ -v
clang version 4.0.1-6 (tags/RELEASE_401/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/7
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/7.2.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.2.0
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/7.2.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
```

**Comment by HowardHinnant** [comment]
I can tell you what is happening, and why it is happening.  But I don't currently have a plan to do anything about it.

Formatting and parsing of dates are implicitly calendrical computations (as opposed to chronological computations).  Calendrical computations involve a calendar (the civil calendar in this example), and chronological computations do not involve calendars, but only durations and time_points.

The calendrical limits can be demonstrated with this code:
```c++
using namespace date;
cout << "year::min() = " << year::min() << '\n';
cout << "year::max() = " << year::max() << '\n';
```
which outputs:
```
year::min() = -32767
year::max() = 32767
```
This means that the calendar can only operate between the years [-32,767, 32,767].

`utc_point` (for example), is a _chronological_ data structure.  It has a far greater range than the civil calendar.  Its approximate range can be shown like this:
```c++
using dyears = duration<double, years::period>;
cout << "utc_point::min() = " << dyears{utc_point::min().time_since_epoch()} << '\n';
cout << "utc_point::max() = " << dyears{utc_point::max().time_since_epoch()} << '\n';
```
which outputs:
```
utc_point::min() = -292277024626.927734[31556952]s
utc_point::max() = 292277024626.927734[31556952]s
```
I.e. `utc_point` has a range of +/- 292 billion years.

I could detect calendrical overflow in the formatting and parsing functions, but this would add a performance penalty for everyone for dubious value.  "Dubious" because although the calendrical range of this library is +/- 32K years, the physical accuracy of the civil calendar in modeling the tropical year of earth is only about 1 day in +/- 4K years.  I.e. once you compute out beyond the year 4,000, mankind will likely have altered the rules of the civil calendar (just as Pope Gregory did in 1582).

So in summary, `utc_point` has a much greater range than this library's calendar, and this library's calendar has a much greater range than our actual civil calendar.  So calendrical computations at these extremes have dubious value.

**Comment by marshray** [comment]
Thanks for the great explanation!
My preference is for functions to raise an exception when they are unable to achieve their postcondition, but then I don't tend to work on applications that do so much date formatting that a bounds check would impose a noticable performance cost.
